src/rgw/rgw_acl_swift.cc

```diff
@@ -192,36 +169,24 @@ int RGWAccessControlPolicy_SWIFT::create(const DoutPrefixProvider *dpp,
   rw_mask = 0;
 
   if (read_list) {
-    std::vector<std::string> uids;
-    int r = parse_list(read_list, uids);
-    if (r < 0) {
-      ldpp_dout(dpp, 0) << "ERROR: parse_list for read returned r="
-                    << r << dendl;
-      return r;
-    }
-
-    r = add_grants(dpp, driver, uids, SWIFT_PERM_READ);
-    if (r < 0) {
-      ldpp_dout(dpp, 0) << "ERROR: add_grants for read returned r="
-                    << r << dendl;
-      return r;
+    for (std::string_view uid : ceph::split(read_list, " ,")) {
+      int r = add_grant(dpp, driver, std::string{uid}, SWIFT_PERM_READ);
```

Comment:
if you allocate an `std::string` anyway, why use a `string_view`?

Reply:
i had tried to use `for (std::string uid :`, but it didn't compile because `ceph::split()` returns a range of string_views which don't implicitly convert to string :( so i found the explicit conversion `std::string{uid}` to be necessary